packages/kbn-ebt-tools/src/metric_events/schema.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { RootSchema } from '@kbn/analytics-client';
+
+/**
+ * Structure of the `metric` event
+ */
+export interface MetricEvent {
+  /**
+   * The name of the action that is tracked in the metrics.
+   */
+  eventName: string;
+  /**
+   * Searchable but not aggregateable metadata relevant to the tracked action.
+   */
+  meta?: Record<string, unknown>;
+
+  /**
+   * @group Standardized fields
+   * The time (in milliseconds) it took to run the entire action.
+   */
+  duration?: number;
+  /**
+   * @group Standardized fields
+   * A status relevant to the action (i.e.: `failed`, `succeeded`).
+   */
+  status?: string;
+  /**
+   * @group Standardized fields
+   * performance.memory.jsHeapSizeLimit
+   */
+  jsHeapSizeLimit?: number;
+  /**
+   * @group Standardized fields
+   * performance.memory.totalJSHeapSize
+   */
+  totalJSHeapSize?: number;
+  /**
+   * @group Standardized fields
+   * performance.memory.usedJSHeapSize
+   */
+  usedJSHeapSize?: number;
```

Comment:
It's not clear to me why we have these "standardised" fields, it doesn't look like we want to send these values for each performance metric event, so shouldn't we rather use a `eventName: "jsHeap"` and use the free fields for the different values we're tracking?

Otherwise we should add "standardised" fields for event loop delay, event loop utilization etc. I don't see a reason to treat any of these metrics differently.

Reply:
IMHO, standardised fields are the fields which can be used at any event (optionally) if the developer wants to report those metrics. We structured free fields to be more like sub-metrics of main event. I think you are pretty much right if you think jsHeap can be event itself and you can use free fields as you mentioned, but I don't really believe jsHeap or any other general concept should be reported in it's own event. For example, It's more interesting to see `jsHeap` relevant standard fields during 'dashboard-load' event than jsHeap event where we have no idea about in which condition/where it's reported.